### PR TITLE
Switch to 1ES templates

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/dotnet/aspnetcore-internal/issues/4468

We unfortunately have to bifurcate our .yml files into a `ci.yml` (internal) and `ci-public.yml` (public) as part of this, which means we have some duplicated code. Would be good to clean some of that up in the future.

Latest test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2409961&view=results

### CODE REVIEW

**Comment by wtgodbe** [comment]
Once the internal build passes I'll switch our public CI over to using the public yml file

File: .azure/pipelines/jobs/default-build.yml [hunk 1]
```diff
@@ -78,151 +78,357 @@ parameters:
...
+      - _BuildConfig: ${{ parameters.configuration }}
+      - BuildConfiguration: ${{ parameters.configuration }}
+      - BuildDirectory: ${{ parameters.buildDirectory }}
+      - DOTNET_CLI_HOME: $(System.DefaultWorkingDirectory)
+      - DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+      - TeamName: AspNetCore
+      - ${{ if eq(parameters.agentOs, 'Linux') }}:
+        - LC_ALL: 'en_US.UTF-8'
+        - LANG: 'en_US.UTF-8'
+        - LANGUAGE: 'en_US.UTF-8'
```
Comment (RussKie) [inline .azure/pipelines/jobs/default-build.yml:136]: These are set unconditionally below

File: .azure/pipelines/jobs/default-build.yml [hunk 2]
```diff
@@ -78,151 +78,357 @@ parameters:
...
+          condition: always()
+          inputs:
+            testResultsFormat: JUnit
+            testResultsFiles: '**/artifacts/log/**/*.junit.xml'
+            testRunTitle: $(AgentOsName)-$(BuildConfiguration)-js
+            mergeTestResults: true
+            buildConfiguration: $(BuildConfiguration)
+            buildPlatform: $(AgentOsName)
+
+- ${{ if eq(variables['System.TeamProject'], 'internal') }}:
```
Comment (RussKie) [inline .azure/pipelines/jobs/default-build.yml:297]: As discussed offline, I would split these steps into own file for the ease of maintenance. I'd name these files something like build-public.yml and buid-internal.yml.

**Review by RussKie — APPROVED** [review]

**Pull request merged**